### ISSUE TITLE
Track proxied calls as dependencies in telemetry

### ISSUE THREAD

**Issue by luk355** [microsoft/reverse-proxy#1480]
### What should we add or change to make your life better?

Track proxied calls as a dependency in telemetry. I would expect this level of tracking to come as a default by YARP the same way as when you use HttpClient.

An appInsights example of how telemetry looks for request proxied by YARP:

<img width="625" alt="image" src="https://user-images.githubusercontent.com/9592890/147250486-d7025761-b668-4187-8e9b-8b965cb27deb.png">

... compared to a request made to a controller action which has injected a HttpClient instance and makes arbitrary http call.

<img width="621" alt="image" src="https://user-images.githubusercontent.com/9592890/147250605-f9b0e359-4a3e-48be-9d17-e9b5ab1cf747.png">

I am experiencing this behavior with .NET 6 & YARP v1.0.

### Why is this important to you?

As YARP is designed to do various transformations, I would be super useful to see the actual HTTP request call made by the proxy.

I was able to find various discussions along this theme, but none of them neither solves them/nor states that this feature should not be part ot the Yarp. See: https://github.com/microsoft/reverse-proxy/issues/708, https://github.com/microsoft/reverse-proxy/pull/456, https://github.com/microsoft/reverse-proxy/discussions/1316

**Comment by MihaZupan** [comment]
Have you seen https://microsoft.github.io/reverse-proxy/articles/distributed-tracing.html?

If you use .NET 6.0, you will see your described behavior (dependency tracking for requests to the downstream) out of the box.
With 3.1/5.0, there is a [workaround](https://microsoft.github.io/reverse-proxy/articles/distributed-tracing.html#net-50-and-older) you can use to enable that functionality before 6.0.

**Comment by luk355** [comment]
This is the behavior what I am getting with .NET 6.0 and YARP v1.0.0. I did see the distributed tracing article - based on the behavior I am experiencing I thought the article is concerned mainly about the population of tracecontext (W3C) headers.

**Comment by MihaZupan** [comment]
Can you share the logs you are seeing on YARP's side?
How are you configuring AppInsights?

This is the expected result from a minimal YARP configuration after adding `services.AddApplicationInsightsTelemetry();` and configuring the `InstrumentationKey`:
![image](https://user-images.githubusercontent.com/25307628/147270314-c4eaa9e4-ffac-4258-ac17-daf0621c6cee.png)

**Comment by luk355** [comment]
I did play more with how we had the proxy configured and figured out the root-cause - it was a mistake on our side. 

We are using `services.AddApplicationInsightsTelemetry();` configuration along with `InstrumentationKey` to plug the integration with AppInsights in.

To explain what was happening, we were using `IForwarderHttpClientFactory.WrapHandler()` method to populate [B3 tracing headers](https://github.com/openzipkin/b3-propagation) and in order to have the SpanId match with the W3C headers, we were also adding W3C tracing headers to the request:

```
// B3 headers
request.Headers.Add("X-B3-TraceId", activity.TraceId.ToHexString());
request.Headers.Add("X-B3-ParentSpanId", activity.ParentSpanId.ToHexString());
request.Headers.Add("X-B3-SpanId", activity.SpanId.ToHexString());
request.Headers.Add("X-B3-Sampled", activity.Recorded ? "1" : "0");

// W3C headers
request.Headers.Add("traceparent", activity.Id);
request.Headers.Add("tracestate", activity.TraceStateString);
``` 

The problem is, once you add the "traceparent" header in, it seems that the `DiagnosticsHandler` does not log the __Dependency__ telemetry at all.

On a side note, I just learned that this was the best way to add B3 tracing headers and utilizing [DistributedContextPropagator](https://github.com/dotnet/runtime/blob/1fd3222cd3ef71708405c2642b5e68cc5053170c/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs) is probably cleaner way to do it. That way, there is no need for custom implementation of IForwarderHttpClientFactory and Dependency tracing works like a charm.

I really appreciate your help @MihaZupan.
